### PR TITLE
MNT Use GitHub Actions CI

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/gha-ci/issues/11<br /><br />Workflow run https://github.com/creative-commoners/cwp/actions?query=branch%3Apulls%2F2.9%2Fmodule-standards

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    - cron: '10 12 * * 1'
+
+jobs:
+  ci:
+    name: CI
+    # Only run cron on the silverstripe account
+    if: (github.event_name == 'schedule' && startsWith(github.repository, 'silverstripe/')) || (github.event_name != 'schedule')
+    uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
+    with:
+      composer_require_extra: cwp/starter-theme:^2
```
Comment (ChloeSartorelli) [inline .github/workflows/ci.yml]: Why not include this as a dev dependency in composer.json instead?

File: composer.json [hunk 2]
```diff
@@ -19,6 +19,9 @@
         "phpunit/phpunit": "^9.5",
         "squizlabs/php_codesniffer": "^3"
     },
+    "conflict": {
+        "silverstripe/auditor": "<2.4.0"
```
Comment (ChloeSartorelli) [inline composer.json]: Why are you adding this conflict? Is there a failed CI run you can point at for example where that specific version was causing issues?

File: tests/PageTypes/DatedUpdateHolderControllerTest.php [hunk 3]
```diff
@@ -54,10 +55,16 @@ public function testInvalidDateFormat()
...
+        try {
+            /** @var EventHolder $holder */
+            $holder = $this->objFromFixture(EventHolder::class, 'EventHolder1');
+            $result = $this->get($holder->Link() . '?from=christmas&to=2018-01-10');
+            // MySQL 5.7
+            $this->assertStringContainsString(htmlentities('Dates must be in "y-MM-dd" format.'), $result->getBody());
+        } catch (DatabaseException $e) {
+            // MySQL 8.0
+            $this->assertStringContainsString("Incorrect DATETIME value: 'christmas 00:00:00'", $e->getMessage());
+        }
```
Comment (ChloeSartorelli) [inline tests/PageTypes/DatedUpdateHolderControllerTest.php]: This difference seems like a bug we should fix, not something we should just consider sane behaviour. We should be catching the `DataBaseException` (or pre-empting it by making sure it doesn't happen in the first place) and returning the same error to the user in the same way we do with MySQL 5.7: https://github.com/silverstripe/cwp/blob/57e38bf69b561cfbcda7ab0429dd88a8d9d8d6a3/src/PageTypes/DatedUpdateHolderController.php#L178-L185

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (emteknetnz) [reply composer.json]: prefer lowest issue
```
1) CWP\CWP\Tests\Extensions\WorkflowDefinitionExtensionTest::testCreateDefaultWorkflowTest
implode(): Passing glue string after array is deprecated. Swap the parameters
/home/runner/work/cwp/cwp/vendor/silverstripe/auditor/code/AuditHook.php:132
```
https://github.com/creative-commoners/cwp/runs/7332407622?check_suite_focus=true#step:12:68

File: .github/workflows/ci.yml (hunk 1 above)
Comment (emteknetnz) [reply .github/workflows/ci.yml]: Updated

File: tests/PageTypes/DatedUpdateHolderControllerTest.php (hunk 3 above)
Comment (emteknetnz) [reply tests/PageTypes/DatedUpdateHolderControllerTest.php]: Will remove and just let the unit test fail, fix it as a regular broken build

File: tests/PageTypes/DatedUpdateHolderControllerTest.php [hunk 4]
```diff
@@ -7,6 +7,7 @@
 use SilverStripe\Dev\FunctionalTest;
 use SilverStripe\View\SSViewer;
 use SilverStripe\ORM\DB;
+use SilverStripe\ORM\Connect\DatabaseException;
```
Comment (ChloeSartorelli) [inline tests/PageTypes/DatedUpdateHolderControllerTest.php]:
```suggestion
```

File: tests/PageTypes/DatedUpdateHolderControllerTest.php [hunk 5]
```diff
@@ -57,6 +58,7 @@ public function testInvalidDateFormat()
         /** @var EventHolder $holder */
         $holder = $this->objFromFixture(EventHolder::class, 'EventHolder1');
         $result = $this->get($holder->Link() . '?from=christmas&to=2018-01-10');
+        // MySQL 5.7
```
Comment (ChloeSartorelli) [inline tests/PageTypes/DatedUpdateHolderControllerTest.php]:
```suggestion
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (ChloeSartorelli) [reply composer.json]: That was resolved in 2.2.1. Please update to `<2.2.1`
https://github.com/silverstripe/silverstripe-auditor/releases/tag/2.2.1

File: tests/PageTypes/DatedUpdateHolderControllerTest.php (hunk 4 above)
Comment (emteknetnz) [reply tests/PageTypes/DatedUpdateHolderControllerTest.php]: Updated

File: tests/PageTypes/DatedUpdateHolderControllerTest.php (hunk 5 above)
Comment (emteknetnz) [reply tests/PageTypes/DatedUpdateHolderControllerTest.php]: Updated

File: composer.json (hunk 2 above)
Comment (emteknetnz) [reply composer.json]: updated

**Review by GuySartorelli — APPROVED** [review]

**Pull request merged**